pandas/core/internals.py

```diff
@@ -1455,15 +1478,52 @@ def should_store(self, value):
         return com.is_integer_dtype(value) and value.dtype == self.dtype
 
 
-class TimeDeltaBlock(IntBlock):
+class DatetimeLikeBlockMixin(object):
+
+    @property
+    def _na_value(self):
+        return tslib.NaT
+
+    @property
+    def fill_value(self):
+        return tslib.iNaT
+
+    def _try_operate(self, values):
+        """ return a version to operate on """
+        return values.view('i8')
+
+    def get_values(self, dtype=None):
+        """
+        return object dtype as boxed values, such as Timestamps/Timedelta
+        """
+        if com.is_object_dtype(dtype):
+            return lib.map_infer(self.values.ravel(),
+                                 self._box_func).reshape(self.values.shape)
+        return self.values
+
+    def quantile(self, values, qs, **kwargs):
+        values = values.view('i8')
+        result = Block.quantile(self, values, qs, **kwargs)
+
+        if com.is_datetime64tz_dtype(self):
+            # ToDo: Temp logic to avoid GH 12619 and GH 12772
+            # which affects to DatetimeBlockTZ_try_coerce_result for np.ndarray
+            if isinstance(result, np.ndarray) and values.ndim > 0:
+                result = self._holder(result, tz='UTC')
```

Comment:
but actually could/should this logic actually be in ``_try_coerce_result`` for ``DatetimeTZBlock``?